File: day4/ruby/day4.rb
require "./day4_lib.rb"

input_min = 156218
input_max = 652527

puts "----- Part 1 -----"

# Brute force

num_passwords = 0

for pass in input_min..input_max do
  if valid_password(pass) then num_passwords += 1 end
end

puts "Total passwords: #{num_passwords}"

puts "----- Part 2 -----"

num_passwords = 0

for pass in input_min..input_max do
  if valid_password_2(pass) then num_passwords += 1 end
end

puts "Total passwords: #{num_passwords}"

File: day04/c++/day4.cpp
#include <iostream>
#include "./day4_lib.h"
#include <fstream>

int main() {

    int min_value = 156218;
    int max_value = 652527;

    std::cout << "----- Part 1 -----" << std::endl;

    int num_passwords = 0;

    for(int pass=min_value; pass<=max_value; pass++){
        if (valid_password(pass)) num_passwords++;
    }

    std::cout << "Total passwords " << num_passwords << std::endl;

    std::cout << "----- Part 2 -----" << std::endl;

    num_passwords = 0;

    for(int pass=min_value; pass<=max_value; pass++){
        if (valid_password_2(pass)) num_passwords++;
    }

    std::cout << "Total passwords " << num_passwords << std::endl;

    std::cout << "----- Part 2 -----" << std::endl;
}
